packages/babel-helpers/src/helpers.ts

```diff
@@ -126,11 +126,18 @@ helpers.asyncGeneratorDelegate = helper("7.0.0-beta.0")`
   export default function _asyncGeneratorDelegate(inner, awaitWrap) {
     var iter = {}, waiting = false;
 
-    function pump(key, value) {
-      waiting = true;
-      value = new Promise(function (resolve) { resolve(inner[key](value)); });
-      return { done: false, value: awaitWrap(value) };
-    };
+    function pump(key, arg) {
+      var step = inner[key](arg);
+      if (step instanceof Promise) {
```

Comment:
I'm really surprised that we need to check if it's a promise and not if it's a thenable.

Reply:
Oh I have no idea. You think thenable is right? It probably is.